### DISCUSSION TITLE
How to open all external links in a new tab

### DISCUSSION THREAD

**Discussion by strboul** [vercel/next.js#21454] [Help] [answered]
I'm trying to find a way to open all the external links in new tabs by using the `<a href="" target="_blank" rel="noreferrer noopener">` attributes. I have a JS script doing that but I couldn't find a way to put it in the Next app that it recognizes right away.

```js
if (typeof window !== "undefined") {
  var links = document.links;
  for (let i = 0, linksLength = links.length; i < linksLength; i++) {
    if (links[i].hostname !== window.location.hostname) {
      links[i].target = "_blank";
      links[i].rel = "noreferrer noopener";
    }
  }
}
```

If I do it on the browser console, it works very well.

I've tried `<script dangerouslySetInnerHTML={{__html: ` inside the `<Head>` tag in the `pages/app.js` but that didn't work.

**Comment by leerob** [comment]
You can create a `Link` component and re-use that throughout your application!

```
const CustomLink = (props) => {
  const href = props.href;
  const isInternalLink = href && (href.startsWith('/') || href.startsWith('#'));

  if (isInternalLink) {
    return (
      <Link href={href}>
        <a {...props} />
      </Link>
    );
  }

  return <a target="_blank" rel="noopener noreferrer" {...props} />;
};
```

**Reply by strboul** [reply]
Thanks for the answer, Lee. I appreciate it.

I forgot to write that my app also has some links inside the MD files, which are created such as `[here](https://example.com)`. For these links, I don't want to use React components, just want to keep them in the `[]()` structure. That's the reason why I tried to achieve this by using the DOM interface.

**Comment by strboul** [comment, marked as answer]
It's possible to override MDX tags with `MDXProvider`. With @leerob's `CustomLink` component, one needs to update the `pages/_app.js` in that way:

```js
import CustomLink '../components/CustomLink';
import { MDXProvider } from '@mdx-js/react';

function MyApp({ Component, pageProps }) {
  return (
    <MDXProvider
      components={{
        a: CustomLink,
      }}
    >
      <Component {...pageProps} />
    </MDXProvider>
  );
}
```

**Comment by benmneb** [comment]
Is it possible to do this without using `MDX`? I'm using `remark` to parse the markdown. 🙏
